### DISCUSSION TITLE
KnowWE build failed

### DISCUSSION THREAD

**Discussion by VitaliyMez** [denkbares/d3web-KnowWE#21] [Q&A] [answered]
Hi, I'm trying to build KnowWE following https://www.d3web.de/Wiki.jsp?page=How-To%20Create%20Your%20Own%20Workspace 

[ERROR] Failed to execute goal on project KnowWE-Plugin-d3web-basic: Could not resolve dependencies for project de.uniwue.d3web:KnowWE-Plugin-d3web-basic:jar:12.11-SNAPSHOT: The following artifacts could not be resolved: de.uniwue.d3web:d3web-Kernel:jar:12.11-SNAPSHOT, de.uniwue.d3web:d3web-Plugin-Kernel-ExtensionPoints:jar:12.11-SNAPSHOT, de.uniwue.d3web:d3web-Plugin-BasicProperties:jar:12.11-SNAPSHOT, de.uniwue.d3web:d3web-Persistence:jar:12.11-SNAPSHOT, de.uniwue.d3web:d3web-Plugin-Persistence-ExtensionPoints:jar:12.11-SNAPSHOT, de.uniwue.d3web:d3web-Plugin-XCL:jar:12.11-SNAPSHOT, de.uniwue.d3web:d3web-Plugin-Persistence-Basic:jar:12.11-SNAPSHOT, de.uniwue.d3web:d3web-Plugin-Interview:jar:12.11-SNAPSHOT, de.uniwue.d3web:d3web-Plugin-SessionPersistence-Basic:jar:12.11-SNAPSHOT, de.uniwue.d3web:d3web-SessionPersistence:jar:12.11-SNAPSHOT, de.uniwue.d3web:d3web-Plugin-SessionPersistence-ExtensionPoints:jar:12.11-SNAPSHOT: Could not find artifact de.uniwue.d3web:d3web-Kernel:jar:12.11-20230113.030139-346 in des-repo-public (https://repo.denkbares.com/repo-public/)

At the same time, all mentioned files look available here: https://repo.denkbares.com/repo-public/de/uniwue/d3web/d3web-Kernel/12.11-SNAPSHOT/ 

Please help.

**Comment by AlbrechtStriffler** [comment, marked as answer]
Hi, I think I need a little bit more info about what you are doing.
What I just tried successfully:

1. Create new Ubuntu-VM
2. Install Java 17 and Maven in Terminal
3. clone https://github.com/denkbares/d3web-KnowWE.git
4. cd to d3web-KnowWE
5. mvn install

There actually really was a wrong dependency to a closed source artifact (not one of those in your trace), but I just fixed that, so it now just works..

![image](https://user-images.githubusercontent.com/13681360/212341820-ba3e0f9a-912c-4014-9496-5430a3e4d5ea.png)

**Comment by VitaliyMez** [comment]
Thank you for your answer!
I had this trace during compilation under Eclipse, but next compiled without errors under IntelliJ IDEA.
